DataLink.tex

```diff
@@ -1061,72 +1076,64 @@ \subsection{Example: Self-Describing Service}
 A service may include a service descriptor that describes itself with
 its normal output. This usage is comparable to prototype work on S3
 (see \citet{note:s3})
-and when combined with calling a service with no input parameters
-(e.g.\ as allowed in \ref{sec:resourceId})
-will make it easy for clients to obtain a
-description of both standard and custom features.
+and will make it easy for clients to obtain a
+description of both standard and custom features (eg PARAMETER Options or specific endpoints for the resource) it gives ranges or options of all accepted parameters. When combined with calling a service with some of the required parameters only (eg an ID)  the response may give defaults specific to the resource identified or selected  by these parameters.
+
+
 
-The output of a \blinks\ capability with no input ID would include the
-self-describing service descriptor and an empty results table:
 \begin{verbatim}
-   <RESOURCE type="meta" utype="adhoc:service" name="this">
-     <PARAM name="standardID" datatype="char" arraysize="*"
-            value="ivo://ivoa.net/std/DataLink#links-1.0" />
-     <PARAM name="accessURL" datatype="char" arraysize="*"
-            value="http://example.com/mylinks" />
-     <GROUP name="inputParams">
-       <PARAM name="ID" datatype="char" arraysize="*"
-              value="" ref="primaryID"/>
-     </GROUP>
-   </RESOURCE>
-   <RESOURCE type="results">
-     <TABLE>
-       <FIELD name="ID" datatype="char" arraysize="*"
-              ucd="meta.id;meta.main" />
-       <FIELD name="access_url" datatype="char" arraysize="*"
-              ucd="meta.ref.url" />
-       <FIELD name="service_def" datatype="char" arraysize="*"
-              ucd="meta.ref" />
-       <FIELD name="error_message" datatype="char" arraysize="*"
-              ucd="meta.code.error" />
-       <FIELD name="semantics" datatype="char" arraysize="*"
-              ucd="meta.code" />
-       <FIELD name="description" datatype="char" arraysize="*"
-              ucd="meta.note" />
-       <FIELD name="content_type" datatype="char" arraysize="*"
-              ucd="meta.code.mime" />
-       <FIELD name="content_length" datatype="long" unit="byte"
-              ucd="phys.size;meta.file" />
-       <DATA> <TABLEDATA/> </DATA>
-     </TABLE>
-   </RESOURCE>
+<RESOURCE type=”meta” utype=”adhoc:service” ID=”PwL” name=”Power Law fitting”>
```

Comment:
In the previous text, the self-describing descriptor specified name="this" so it could be unambiguously distinguished from descriptors for other services. Admittedly, DataLink-1.0 says that is a convention (~SHOULD) rather than a MUST, so we could let services use the name attribute... but we still need a way to separate the self-describing descriptor from other descriptors. 

The only other attribute at our disposal is utype: what about utype="adhoc:this" or "adhoc:this-service"?? I kind of like adhoc:this as it says "resource is meta about this" and doesn't clobber use of name.